src/main/java/Control/Scenario/UC_Graph.java

```diff
@@ -0,0 +1,163 @@
+        } catch (SQLException sqle) {
+            CustomLogs.Error(sqle.getMessage());
+        }
+
+        return arr;
+    }
+
+    public List<Sensor> getSensorsForFlat(Integer flatID) {
+        List<Sensor> sensors = new ArrayList<>();
+
+        List<T_ControllerUnit> t_controllerUnits = getAllControllersForFlat(flatID);
+
+        for (T_ControllerUnit t_controllerUnit : t_controllerUnits) {
+            List<T_Sensor> t_sensors = getAllSensorsForController(t_controllerUnit.getA_pk());
+            for (T_Sensor t_sensor : t_sensors) {
+                try {
+                    List<T_Measurement> t_measurements = I_Measurements.getLast30DaysMeasurements(db.getConn(), db.getPs(), db.getRs(), t_sensor.getA_pk());
+                    List<Measurement> measurements = new ArrayList<>();
+                    for (T_Measurement t_measurement : t_measurements) {
+                        Measurement measurement = new Measurement(t_measurement.getA_AccumulatedValue(), t_measurement.getA_SensorID(), t_measurement.getA_MeasuredAt());
+                        measurements.add(measurement);
+                    }
+
+                    List<Integer> dataArray = getMeasurementArrayForSensor(t_sensor.getA_pk(), measurements);
+
+                    Sensor sensor = new Sensor(t_sensor.getA_pk(), t_sensor.getA_Input(), t_sensor.getA_Name(), t_sensor.getA_SensorTypeID(), t_sensor.getA_ControllerUnitID(), measurements, dataArray);
+                    sensors.add(sensor);
+                } catch (SQLException sqle) {
+                    CustomLogs.Error(sqle.getMessage());
+                }
+            }
+        }
+        return sensors;
+    }
+
+    public List<Integer> getMeasurementArrayForSensor(int sensorID, List<Measurement> measurements) {
+        try {
+            List<Integer> dataArray = new ArrayList<>();
+            Integer accumulatedValue30DaysAgo = I_Measurements.getAccumulatedValueOf30DaysAgo(db.getConn(), db.getPs(), db.getRs(), sensorID);
+
+            HashMap<Date, Integer> hashMap = new HashMap<>();
+            for (Measurement measurement : measurements) {
+                if(hashMap.containsKey(measurement.getMeasuredAt())){
+                    if(hashMap.get(measurement.getMeasuredAt()) < measurement.getValue()-accumulatedValue30DaysAgo){
+                        hashMap.put(measurement.getMeasuredAt(), measurement.getValue()-accumulatedValue30DaysAgo);
+                    } //efektivnejsie by bolo odpocitavat akumulovanu hodnotu az pri datumoch
+                }
+                else {
+                    hashMap.put(measurement.getMeasuredAt(), measurement.getValue()-accumulatedValue30DaysAgo);
+                }
+            }
+
+            int fillerValue = 0;
+
+            List<Date> dates = getDatesOfLast30Days();
+            for(Date date : dates){
+                if(hashMap.containsKey(date)){
+                    fillerValue = hashMap.get(date);
+                    //dataArray.add(fillerValue);
```

Comment:
co to preco zakomentovane, bud to tam je alebo nie :D

Reply:
to som tu zabudla, musim to dat prec :D